SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialWorkingSetSubsystem.cpp

```diff
@@ -0,0 +1,199 @@
+#include "Interop/SpatialWorkingSetSubsystem.h"
+
+#include <Actor.h>
+
+#include "Algo/Transform.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/WorkingSetsHandler.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/ViewCoordinator.h"
+
+DEFINE_LOG_CATEGORY_STATIC(LogSpatialActorSetSubsystem, Display, All);
+
+static bool TryParseActorEntityId(const AActor* Actor, const USpatialPackageMapClient& PackageMap, const TCHAR* ActorInfo,
+								  Worker_EntityId& OutEntityId)
+{
+	if (!IsValid(Actor))
+	{
+		UE_LOG(LogSpatialActorSetSubsystem, Warning, TEXT("Actor is invalid. %s"), ActorInfo);
+
+		return false;
+	}
+	const Worker_EntityId ActorEntityId = PackageMap.GetEntityIdFromObject(Actor);
+	if (ActorEntityId == SpatialConstants::INVALID_ENTITY_ID)
+	{
+		UE_LOG(LogSpatialActorSetSubsystem, Warning, TEXT("Actor doesn't have an entity ID. %s Actor: %s"), ActorInfo, *Actor->GetName());
+		return false;
+	}
+
+	if (!Actor->IsActorReady())
+	{
+		UE_LOG(LogSpatialActorSetSubsystem, Warning, TEXT("Actor isn't ready. %s Actor: %s EntityId: %lld"), ActorInfo, *Actor->GetName(),
+			   ActorEntityId);
+		return false;
+	}
+	OutEntityId = ActorEntityId;
+	return true;
+}
+
+static bool TryParseWorkingSetsEntityIds(TArray<AActor*> Members, AActor* Leader, const USpatialPackageMapClient& PackageMap,
+										 SpatialGDK::FWorkingSetState& InOutRequest)
+{
+	bool bIsRequestValid = true;
+
+	InOutRequest.MemberEntities.Empty(Members.Num());
+	for (const AActor* Member : Members)
+	{
+		Worker_EntityId MemberEntityId;
+		if (TryParseActorEntityId(Member, PackageMap, TEXT("Member"), MemberEntityId))
+		{
+			InOutRequest.MemberEntities.Emplace(MemberEntityId);
+		}
+		else
+		{
+			bIsRequestValid = false;
+		}
+	}
+
+	if (!TryParseActorEntityId(Leader, PackageMap, TEXT("Leader"), InOutRequest.LeaderEntityId))
+	{
+		bIsRequestValid = false;
+	}
+
+	return bIsRequestValid;
+}
+
+bool USpatialActorSetSubsystem::IsSetHandleValid(FActorSetHandle SetHandle, UObject* Context)
+{
+	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(Context->GetWorld()->GetNetDriver());
+
+	return SetHandle.ActorSetEntityHandle != SpatialConstants::INVALID_ENTITY_ID
+		   && NetDriver->WorkingSetData->HasMarkerEntity(SetHandle.ActorSetEntityHandle);
+}
+
+namespace SpatialGDK
+{
+static FWorkingSetChangesHandler::FOnWorkingSetEvent ConvertBPDelegate(FOnWorkingSetEventDynamic BlueprintDelegate)
```

Comment:
why include this in the namespace? for readability, would rather this alongside the other static functions at the top of this file

Reply:
Honestly, it's just to save on mental resources while trying to read `SpatialGDK::FWorkingSetChangesHandler::FOnWorkingSetEvent` everywhere. Will move up.